src/lamp_py/ingestion/light_rail_gps.py

```diff
@@ -0,0 +1,225 @@
+                    pl.col("data").struct.field("speed").cast(pl.Float64),
+                    (
+                        pl.col("data")
+                        .struct.field("updated_at")
+                        .str.slice(0, length=10)
+                        .str.to_date()
+                        .alias("date")
+                    ),
+                    pl.col("data").struct.field("updated_at").cast(pl.String),
+                    pl.col("data").struct.field("bearing").cast(pl.Float64),
+                    pl.col("data").struct.field("latitude").cast(pl.String),
+                    pl.col("data").struct.field("longitude").cast(pl.String),
+                    pl.col("data").struct.field("car").cast(pl.String),
+                )
+            )
+
+        logger.log_complete()
+
+    except Exception as exception:
+        logger.log_failure(exception)
+        df = None
+
+    return (df, path)
+
+
+def dataframe_from_gz(
+    files: List[str],
+) -> Tuple[pl.DataFrame, List[str], List[str]]:
+    """
+    create polars dataframe from list of file paths using ThreadPool
+
+    :param files: list of gzip file paths to convert to dataframe
+
+    :return
+    dataframe:
+        serial_number: String,
+        speed: Float64,
+        date: Date,
+        updated_at: String,
+        bearing: Float64,
+        latitude: String,
+        longitude: String,
+        car: String,
+
+    archive_files: correctly loaded gzip paths
+    error_files: gzip paths that failed to load
+    """
+    logger = ProcessLogger(
+        process_name="light_rail_df_from_gz", num_files=len(files)
+    )
+    logger.log_start()
+
+    init_file = files[0]
+    dfs = []
+    archive_files = []
+    error_files = []
+
+    try:
+        with ThreadPoolExecutor(
+            max_workers=16, initializer=thread_init, initargs=(init_file,)
+        ) as pool:
+            for df, path in pool.map(thread_gps_to_frame, files):
+                if df is not None:
+                    archive_files.append(path)
+                    dfs.append(df)
+                else:
+                    error_files.append(path)
+
+        dataframe: pl.DataFrame = pl.concat(dfs)
+
+        logger.add_metadata(
+            num_archive_files=len(archive_files),
+            num_error_files=len(error_files),
+        )
+        logger.log_complete()
+
+    except Exception as exception:
+        error_files += files
+        archive_files = []
+        dataframe = pl.DataFrame(schema=raw_gps_schema)
+        logger.log_failure(exception)
+
+    return (dataframe, archive_files, error_files)
+
+
+def write_parquet(dataframe: pl.DataFrame) -> None:
+    """
+    sync and write parquet files partitioned by date column
+
+    will merge any existing parquet files in memory
+    """
+    for date in dataframe.get_column("date").unique():
+        logger = ProcessLogger(
+            process_name="light_rail_write_parquet", date=date
+        )
+        logger.log_start()
```

Comment:
Write parquet files by date pulled from "updated_at" column. 

This will merge existing parquet files from S3, if they exist and de-dupe the entire dataframe.